developer-docs-site/docs/concepts/basics-events.md

```diff
@@ -22,5 +22,5 @@ Each registered event has a unique `key`. The key `0x0000000000000000c40f1c9b9fd
 
 These represent event streams, or a list of events with each entry containing a sequentially increasing `sequence_number` beginning at `0`, a `type`, and `data`. Each event must be defined by some `type`. There may be multiple events defined by the same or similar `type`s especially when using generics. Events have associated `data`. The general principle is to include all data necessary to understand the changes to the underlying resources before and after the execution of the transaction that changed the data and emitted the event.
 
-[coin_transfer]: https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/Coin.move#L304
-[get_events]: https://fullnode.devnet.aptoslabs.com/spec.html#/operations/get_events_by_event_handle
+[coin_transfer]: https://github.com/aptos-labs/aptos-core/blob/bdd0a7fe82cd6aab4b47250e5eb6298986777cf7/aptos-move/framework/aptos-framework/sources/coin.move#L412
```

Comment:
Let's not ping to a specific commit, can you change it back to `main`? Either that or `devnet`.

Reply:
if we don't pin to a specific commit, then the line numbers won't line up. I've started pinning to specific commits due to this :/

Reply:
Actually this is the permalink. Isn't it always safe to use the permalink to the line? Or are we talking two different things?

Reply:
David's point is if we link to a line number but not to a specific commit, as the file changes, the line could point to something else. So that's a good point, let's leave it the way you have it, with the specific commit.